app/models/volunteer.rb

```diff
@@ -3,4 +3,13 @@ class Volunteer < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+  has_many :assignments
+  has_many :tasks, :through => :assignments
+  has_many :events, :through => :tasks
+  before_destroy :destroy_assignments
+
+  def destroy_assignments
```

Comment:
you should be able to do something like in events like:

`has_many :assignments, dependent: destroy`

in fact, all your has_many relationships (as far as I can tell) should have destroy be dependent.